iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/ActivePowerControlImpl.java

```diff
@@ -43,22 +45,54 @@ public ActivePowerControlImpl(T component,
                                   boolean participate,
                                   double droop,
                                   double participationFactor,
-                                  double minPOverride,
-                                  double maxPOverride) {
+                                  double minTargetP,
+                                  double maxTargetP) {
         super(component);
         int variantArraySize = getVariantManagerHolder().getVariantManager().getVariantArraySize();
         this.participate = new TBooleanArrayList(variantArraySize);
         this.droop = new TDoubleArrayList(variantArraySize);
         this.participationFactor = new TDoubleArrayList(variantArraySize);
-        this.minPOverride = new TDoubleArrayList(variantArraySize);
-        this.maxPOverride = new TDoubleArrayList(variantArraySize);
-        this.allTDoubleArrayLists = List.of(this.droop, this.participationFactor, this.minPOverride, this.maxPOverride);
+        this.minTargetP = new TDoubleArrayList(variantArraySize);
+        this.maxTargetP = new TDoubleArrayList(variantArraySize);
+        this.allTDoubleArrayLists = List.of(this.droop, this.participationFactor, this.minTargetP, this.maxTargetP);
         for (int i = 0; i < variantArraySize; i++) {
             this.participate.add(participate);
             this.droop.add(droop);
             this.participationFactor.add(participationFactor);
-            this.minPOverride.add(minPOverride);
-            this.maxPOverride.add(maxPOverride);
+            this.minTargetP.add(checkTargetPLimit(minTargetP, "minTargetP", component));
+            this.maxTargetP.add(checkTargetPLimit(maxTargetP, "maxTargetP", component));
+        }
+        checkTargetPLimits(minTargetP, maxTargetP);
+    }
+
+    private double checkTargetPLimit(double targetPLimit, String name, T injection) {
```

Comment:
The checks are made:
- on extension creation time
- on extension value change

However if the user modifies the extendable's (Generator or Battery) minP or maxP, we may end up in a situation where minTargetP/maxTargetP is not contained anymore within minP / maxP.

We could:
- consider this as a marginal use case and leave this as it is (or rather: just document this)
- modify getMinTargetP() and getMaxTargetP() to account for minP and maxP. This would probably be the easiest solution.
- add listener, get notifications of minP and maxP change, and in case of check not passing:
  - automatically correct minTargetP/maxTargetP to stay within minP/maxP change (much more user friendly)
  - restore previous minP/maxP and throw (much less user friendly)

Reply:
Solution 2 done, with a warning. One advantage is that you are guaranteed to be able to reload a serialized file.